honeybee_radiance/modifierset.py

```diff
@@ -0,0 +1,702 @@
+                if getattr(self, attr) is not None else None
+                for attr in attributes
+                }
+        else:
+            base = {
+                attr[1:]:getattr(self, attr[1:]).name
+                for attr in attributes
+            }
+        
+        base['type'] = self.__class__.__name__ + 'Abridged'
+        return base
+
+    def duplicate(self):
+        """Get a copy of this set."""
+        return self.__copy__()
+
+    def _validate_modifier(self, value):
+        """Check an modifier before assigning it."""
+        if value is not None:
+            assert isinstance(value, Primitive) and value.can_be_modifier , \
+                'Expected modifier. Got {}'.format(type(value))
+            # TODO: This should be uncommented once this is addressed
+            # 
+            # value.lock()   # lock editing in case modifier has multiple references
+        return value
+
+    @property
+    def _slots(self):
+        """Return slots including the ones from the baseclass if any."""
+        slots = set(self.__slots__)
+        for cls in self.__class__.__mro__[:-1]:
+            for s in getattr(cls, '__slots__', tuple()):
+                if s in slots:
+                    continue
+                slots.add(s)
+        slots.remove('_locked')
+        return slots
+
+    def ToString(self):
+        """Overwrite .NET ToString."""
+        return self.__repr__()
+
+    def __len__(self):
+        return 2
+
+    def __iter__(self):
+        return iter(self.modifiers)
+
+    def __copy__(self):
+        return self.__class__(self._exterior_modifier, self._interior_modifier)
+
+    def __repr__(self):
+        name = self.__class__.__name__.split('Set')[0]
+        return '{} Modifier Set:\n Exterior: {}\n Interior: {}'.format(
+            name, self.exterior_modifier.name, self.interior_modifier.name
+        )
+
+
+@lockable
+class WallSet(_BaseSet):
+    """Set containing all radiance modifiers needed to for an radiance model's Walls.
+
+    Properties:
+        exterior_modifier
+        interior_modifier
+        modifiers
+        modified_modifiers
+        is_modified
+    """
+    __slots__ = ()
+
+
+@lockable
+class FloorSet(_BaseSet):
+    """Set containing all radiance modifiers needed to for an radiance model's floors.
+
+    Properties:
+        exterior_modifier
+        interior_modifier
+        modifiers
+        modified_modifiers
+        is_modified
+    """
+    __slots__ = ()
+
+
+@lockable
+class RoofCeilingSet(_BaseSet):
+    """Set containing all radiance modifiers needed to for an radiance model's roofs.
+
+    Properties:
+        exterior_modifier
+        interior_modifier
+        modifiers
+        modified_modifiers
+        is_modified
+    """
+    __slots__ = ()
+
+
+@lockable
+class ApertureSet(_BaseSet):
+    """Set containing all radiance modifiers needed to for a radiance model's Apertures.
+
+    Properties:
+        exterior_modifier
```

Comment:
Actually now that you posted this it got me thinking that and Aperture by definition cannot be interior so the logical move is to remove the interior one but then where do we address interior windows?

The window from aperture distinction has started to be an issue. Not sure what is the best solution here but what I think should not change is that an aperture is an exterior window, skylight or glass door.

>  But I also realize that, as a developer, you are probably in love with the beauty of inherited code as I would be had I written it.

[?] - not sure if I follow all the love and feeling here! 😕 I think once we can sort out the distinction between interior window and aperture this will be sorted out by default. Maybe I should double check how you handled it for `honeybee-energy`.

Reply:
Maybe it should be a WindowSet instead of ApertureSet?